rust/chains/hyperlane-cosmos/src/mailbox.rs

```diff
@@ -376,15 +376,18 @@ impl Indexer<HyperlaneMessage> for CosmosMailboxIndexer {
             .await
             .into_iter()
             .flatten()
-            .filter_map(|(logs_res, block_number)| match logs_res {
-                Ok(logs) => Some(logs),
-                Err(err) => {
+            .map(|(logs, block_number)| {
+                if let Err(err) = &logs {
                     warn!(?err, ?block_number, "Failed to fetch logs for block");
-                    None
                 }
+                logs
             })
+            // Propagate errors from any of the queries. This will cause the entire range to be retried,
+            // including successful ones, but we don't have a way to handle partial failures in a range for now.
+            .collect::<Result<Vec<_>, _>>()?
+            .into_iter()
             .flatten()
-            .map(|(log, meta)| (log.into(), meta))
```

Comment:
I just copied over the existing implementation from the IGP indexer, this line does the same thing as `.into()` but not worth changing back now